dspace-api/src/test/java/org/dspace/discovery/DiscoveryIT.java

```diff
@@ -633,22 +636,90 @@ public void disabledRerunOfSolrQueryDueToStaleObjectsTest() throws Exception {
-        assertSearchQuery(IndexableItem.TYPE, 3, 3, 0, -1);
+        assertSearchQuery(IndexableItem.TYPE, 3, 3, 0, - 1);
+    }
+
+    @Test
+    public void iteratorSearchServiceTest() throws SearchServiceException {
+        String subject1 = "subject1";
+        String subject2 = "subject2";
+        int numberItemsSubject1 = 30;
+        int numberItemsSubject2 = 2;
+        Item[] itemsSubject1 = new Item[numberItemsSubject1];
+        Item[] itemsSubject2 = new Item[numberItemsSubject2];
+        context.turnOffAuthorisationSystem();
+        try {
+            Community community = CommunityBuilder.createCommunity(context).build();
+            Collection collection = CollectionBuilder.createCollection(context, community).build();
+            for (int i = 0; i < numberItemsSubject1; i++) {
+                itemsSubject1[i] = ItemBuilder.createItem(context, collection)
+                    .withTitle("item subject 1 number" + i)
+                    .withSubject(subject1)
+                    .build();
+            }
+
+            for (int i = 0; i < numberItemsSubject2; i++) {
+                itemsSubject2[i] = ItemBuilder.createItem(context, collection)
+                    .withTitle("item subject 2 number " + i)
+                    .withSubject(subject2)
+                    .build();
+            }
+
+            Collection collection2 = CollectionBuilder.createCollection(context, community).build();
+            ItemBuilder.createItem(context, collection2)
+                .withTitle("item collection2")
+                .withSubject(subject1)
+                .build();
+
+            DiscoverQuery discoverQuery = new DiscoverQuery();
+            discoverQuery.addFilterQueries("subject:" + subject1);
+
+            Iterator<Item> itemIterator =
+                searchService.iteratorSearch(context, new IndexableCollection(collection), discoverQuery);
+            int counter = 0;
+            List<Item> foundItems = new ArrayList<>();
+            while (itemIterator.hasNext()) {
+                foundItems.add(itemIterator.next());
+                counter++;
+            }
+            for (Item item : itemsSubject1) {
+                assertTrue(foundItems.contains(item));
+            }
+            assertEquals(numberItemsSubject1, counter);
+
+            discoverQuery = new DiscoverQuery();
+            discoverQuery.addFilterQueries("subject:" + subject2);
+
+            itemIterator = searchService.iteratorSearch(context, null, discoverQuery);
+            counter = 0;
+            foundItems = new ArrayList<>();
+            while (itemIterator.hasNext()) {
+                foundItems.add(itemIterator.next());
+                counter++;
+            }
+            assertEquals(numberItemsSubject2, counter);
+            for (Item item : itemsSubject2) {
+                assertTrue(foundItems.contains(item));
+            }
+
+        } finally {
+            context.restoreAuthSystemState();
```

Comment:
We should not leave authorization turned off during tests...as it can result in unexpected behavior (especially since you use this `context` in your Tests to call `iteratorSearch`...which means this search is always performed with auth checks disabled).  Please move this line up to just after you create your test content, e.g. around line 680 in this file.

Reply:
This has been moved